File: tree.py
class Node(object):
	def __init__(self, data=None, left=None, right=None):
		self.left=left
		self.right=right
		self.data=data

def freeTree(root):
	if root == None:
		return None
	root.left = freeTree(root.left)
	root.right = freeTree(root.right)
	return None

def getLeaves(root, leaves):
	if root == None:
		return
	elif root.left != None or root.right != None:
		getLeaves(root.left, leaves)
		getLeaves(root.right, leaves)
	else:
		leaves.append(root)
	return leaves

File: tree.c
#include "tree.h"

treeNode* freeTree(treeNode* root) {
				if (root == NULL) {
								return NULL;
				}
				root->left = freeTree(root->left);
				root->right = freeTree(root->right);
				return NULL;
}

treeNode** getLeaves(treeNode* root) {
				if (root == NULL) {
								return NULL;
				}
				if (root->left != NULL or root->right != NULL) {
								/* getLeaves of left and right and combine them */
				}
				/* root has no left or right, so it is a leaf */
				return &root;
}
